### PR TITLE
Disable holdout set in AutoML search by default

### PR DESCRIPTION
**Pull request by christopherbunn**
Unblocking next EvalML release, will need to renable once we investigate why it is erroring out with the oversampler.

### CODE REVIEW

File: evalml/automl/automl_search.py [hunk 1]
```diff
@@ -1724,6 +1724,7 @@ def train_pipelines(self, pipelines):
             if computation.done():
                 try:
                     fitted_pipeline = computation.get_result()[0]
+                    self.logger.info(fitted_pipeline.name)
```
Comment (jeremyliweishih) [inline evalml/automl/automl_search.py:1727]: Let's remove this

**Review by chukarsten — APPROVED** [review]

**Review by jeremyliweishih — APPROVED** [review]
LGTM once tests pass. We should cleanup that debugging log though. @christopherbunn just to make sure - is this the only logic we need to change? It looks like it to me. Let's check the docs as well to see if our examples still make sense with this change.

**Comment by christopherbunn** [comment]
@jeremyliweishih I'll update the docs to reflect our new default behavior

**Comment by christopherbunn** [comment]
https://github.com/alteryx/evalml/pull/3659/commits/a652d2dd006748e1bcd6f35fb0b410ea395f6e6e has the tests and docs prior to the adjustments made to pass the tests

**Pull request merged**